prime-dotnet-webapi/Services/Rules/MinorUpdateRules.cs

```diff
@@ -23,14 +24,57 @@ public abstract class MinorUpdateRule : IEnrolleeRule
     /// </summary>
     public class CurrentToaRule : MinorUpdateRule
     {
+        private readonly IEnumerable<int> _newestAgreementVersionIds;
+
+        public CurrentToaRule(IEnumerable<int> newestAgreementVersionIds)
+        {
+            _newestAgreementVersionIds = newestAgreementVersionIds;
+        }
         public override Task<bool> ProcessRule(Enrollee enrollee)
         {
             if (enrollee.Agreements == null)
             {
                 return Task.FromResult(false);
             }
 
-            return Task.FromResult(enrollee.HasLatestAgreement());
+            return Task.FromResult(enrollee.HasLatestAgreement(_newestAgreementVersionIds));
+        }
+    }
+
+    /// <summary>
+    /// Enrollee has a signed a TOA that is of the type the agreementEngine would assign
+    /// at this point in time. In the case the agreementEngine returns indetermanate, return false.
+    /// </summary>
+    public class CorrectToaRule : MinorUpdateRule
+    {
+        public override Task<bool> ProcessRule(Enrollee enrollee)
```

Comment:
Looking at the calling stack upward, it's called by using `await`. should it have modifier `async`?

Reply:
Short answer no.

Long answer is you only require async on a method when there is a await being used inside the method. that will then wrap the method in Task. You will get a warning when adding the async to these methods. Also you **cannot** return Tasks inside an async method since it does the wrapping itself. 

From what I gather is since all these rules are done sync this is kind of the way to wrap the sync methods to return a Task. In the case a rule would have to return something from an async call you would need to make the method async and not wrap that. But the plan was to reduce the amount of dependancies in the Rule files from what i gathered.